sunpy/map/tests/test_mapbase.py

```diff
@@ -1859,3 +1860,16 @@ def test_set_wcs_modifies_crpix(aia171_test_map, aslice, dims):
                                                               aia171_test_map.meta["CRPIX2"])
 
     assert np.allclose(sliced_ref_coord, ori_ref_coord)
+
+def test_slicing_map_submap(aia171_test_map):
+    bl_coord = SkyCoord(-1000 * u.arcsec, 0 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    bottom_left = aia171_test_map.wcs.world_to_pixel(bl_coord)
+
+    tp_coord = SkyCoord(0 * u.arcsec, 1000 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    top_right = aia171_test_map.wcs.world_to_pixel(tp_coord)
+
+    submap = aia171_test_map.submap(bottom_left*u.pix, top_right=top_right*u.pix)
+    sliced_map = aia171_test_map[int(bottom_left[1]):int(top_right[1]) + 1,
+                                 int(bottom_left[0]) +1:int(top_right[0]) + 2]
```

Comment:
If you look at what I had to add to the slicing for the submap to be the same, something feels off.